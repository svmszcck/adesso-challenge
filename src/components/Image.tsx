import React, { useState, memo } from "react";
import { View, Image, StyleSheet, StyleProp, ImageStyle } from "react-native";
import { LinearGradient } from "expo-linear-gradient";

import { checkURL } from "@/utils/validation";
import Placeholder from "@/assets/images/placeholder.jpg";

type CustomImageProps = {
  src?: string;
  withGradient?: boolean;
  style?: StyleProp<ImageStyle>;
};

const CustomImage: React.FC<CustomImageProps> = ({
  src,
  style,
  withGradient,
}) => {
  const [loading, setLoading] = useState(true);
  const isValidURL = src ? checkURL(src) : false;

  return (
    <View>
      {loading && (
        <Image
          source={Placeholder}
          style={[styles.placeholder, style]}
          resizeMode="cover"
          testID="placeholder-image"
          accessibilityLabel="Varsayılan resim"
          accessible
        />
      )}
      <Image
        source={isValidURL ? { uri: src } : Placeholder}
        style={style}
        onLoadStart={() => setLoading(true)}
        onLoadEnd={() => setLoading(false)}
        defaultSource={Placeholder}
        resizeMode="cover"
        testID="custom-image"
        accessibilityLabel={isValidURL ? "Yüklenen resim" : "Varsayılan resim"}
        accessible
      />
      {withGradient && isValidURL && (
        <LinearGradient
          colors={["rgba(0, 0, 0, 0.2)", "rgba(0, 0, 0, 0.8)"]}
          style={styles.overlay}
          testID="gradient"
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  placeholder: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
  },
  overlay: {
    position: "absolute",
    bottom: 0,
    width: "100%",
    height: "100%",
  },
});

export default CustomImage;
