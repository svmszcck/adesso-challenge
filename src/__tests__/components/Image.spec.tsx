import { render, screen, userEvent } from "@testing-library/react-native";
import Image from "../../components/Image";

test("form submits two answers", async () => {
  const questions = ["q1", "q2"];
  const onSubmit = jest.fn();

  const user = userEvent.setup();
  render(<Image src="https:/www.ertert.com" />);

  expect(true).toBeTruthy();
});
