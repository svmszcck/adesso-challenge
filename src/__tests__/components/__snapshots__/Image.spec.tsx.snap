// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches snapshot 1`] = `
<View>
  <Image
    accessibilityLabel="Varsayılan resim"
    accessible={true}
    resizeMode="cover"
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
    testID="placeholder-image"
  />
  <Image
    accessibilityLabel="Yüklenen resim"
    accessible={true}
    defaultSource={1}
    onLoadEnd={[Function]}
    onLoadStart={[Function]}
    resizeMode="cover"
    source={
      {
        "uri": "http://valid.url/image.jpg",
      }
    }
    testID="custom-image"
  />
</View>
`;
